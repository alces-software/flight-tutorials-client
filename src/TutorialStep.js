import React, {Component} from 'react';

export default class TutorialStep extends Component {
  render() {
    const { current, completed, step } = this.props;
    if (!current && !completed) {
      return null;
    }
    return (
      <div>
        <h3>{step.title}</h3>
        <div dangerouslySetInnerHTML={{__html: step.description}} />
      </div>
    );
  }
}
