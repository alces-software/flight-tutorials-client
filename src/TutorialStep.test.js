// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import TutorialStep from './TutorialStep';

const step = {
  title: 'My step 1',
  description: 'My step 1 description',
};

const varieties = {
  currentStep: { current: true, completed: false },
  previousStep: { current: false, completed: true },
  futureStep: { current: false, completed: false },
};

Object.keys(varieties).forEach((variety) => {
  it(`renders without crashing (${variety})`, () => {
    const node = document.createElement('div');
    render(<TutorialStep {...varieties[variety]} step={step} />, node);
  });

  it(`renders correctly (${variety})`, () => {
    const tree = renderer.create(
      <TutorialStep {...varieties[variety]} step={step} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Current step', () => {
  it('includes the title', () => {
    const wrapper = shallow(
      <TutorialStep {...varieties.currentStep} step={step} />,
    );

    expect(wrapper).toIncludeText(step.title);
  });

  xit('includes the description', () => {
    const wrapper = shallow(
      <TutorialStep {...varieties.currentStep} step={step} />,
    );

    expect(wrapper).toIncludeText(step.description);
  });
});

describe('Previous steps', () => {
  it('includes the title', () => {
    const wrapper = shallow(
      <TutorialStep {...varieties.previousStep} step={step} />,
    );

    expect(wrapper).toIncludeText(step.title);
  });

  xit('includes the description', () => {
    const wrapper = shallow(
      <TutorialStep {...varieties.previousStep} step={step} />,
    );

    expect(wrapper).toIncludeText(step.description);
  });
});

describe('Future steps', () => {
  it('does not include the title', () => {
    const wrapper = shallow(
      <TutorialStep {...varieties.futureStep} step={step} />,
    );

    expect(wrapper).not.toIncludeText(step.title);
  });

  xit('does not include the description', () => {
    const wrapper = shallow(
      <TutorialStep {...varieties.futureStep} step={step} />,
    );

    expect(wrapper).not.toIncludeText(step.description);
  });
});
