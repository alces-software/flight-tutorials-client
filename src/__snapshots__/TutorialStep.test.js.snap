// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly (currentStep) 1`] = `
<div>
  <h3>
    My step 1
  </h3>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "My step 1 description",
      }
    }
  />
</div>
`;

exports[`renders correctly (futureStep) 1`] = `null`;

exports[`renders correctly (previousStep) 1`] = `
<div>
  <h3>
    My step 1
  </h3>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "My step 1 description",
      }
    }
  />
</div>
`;
